contracts/strategies/morpho/MetaMorphoStrategy.sol

```diff
@@ -256,21 +278,24 @@ contract MetaMorphoStrategy is ERC4626Strategy {
       _market = MORPHO_BLUE.market(_marketId);
       // get vault position data in the target market
       _pos = MORPHO_BLUE.position(_marketId, address(_mmVault));
+      // calc how much of `add` will be added to this market
+      if (_aprData.add > 0) {
+        _aprData.add = _calcMarketAdd(_mmVault, _marketId, _aprData.supplyLen, _aprData.add);
+      }
+      // calc how much of `sub` will be added to this market
+      if (_aprData.sub > 0) {
+        _aprData.sub = _calcMarketSub(_mmVault, _marketId, _aprData.withdrawLen, _aprData.sub);
+      }
       // get underlyings supplied by the vault in the target market
       // totalSupplyShares : totalSupplyAssets = supplyShares : assetsSuppliedByVault
       // => assetsSuppliedByVault = supplyShares * totalSupplyAssets / totalSupplyShares
-      _assetsSuppliedByVault = _pos.supplyShares * _market.totalSupplyAssets / _market.totalSupplyShares;
-      // get % (in EXP_SCALE) of vault assets that will go in this specific market once deposited (not all assets will go in the same market)
-      _currPercOfAssetsForMarket = _assetsSuppliedByVault * EXP_SCALE / _aprData.totalAssets;
-      // and scale add and sub values to maintain the proportion, calculates with vault totalAssets and assets supplied by vault
-      _aprData.add = _aprData.add * _currPercOfAssetsForMarket / EXP_SCALE;
-      _aprData.sub = _aprData.sub * _currPercOfAssetsForMarket / EXP_SCALE;
+      _assetsSuppliedByVault = _pos.supplyShares * _market.totalSupplyAssets / _market.totalSupplyShares + _aprData.add - _aprData.sub;
       // calculate new totalSupplyAssets with liquidity added/removed
-      _newTotalSupplyAssets = _market.totalSupplyAssets + _aprData.add - _aprData.sub;
+      _totalSupplyAssets = _market.totalSupplyAssets + _aprData.add - _aprData.sub;
       // calculate vaultShare (% in EXP_SCALE) of the total market and simulate change of liquidity by using add and sub
-      _vaultShare = (_assetsSuppliedByVault + _aprData.add - _aprData.sub) * EXP_SCALE / _newTotalSupplyAssets;
+      _vaultShare = _assetsSuppliedByVault * EXP_SCALE / _totalSupplyAssets;
```

Comment:
do we agree that those shares should be different than the balance of vault shares right?

Reply:
Yes, here I'm just calculating the % of assets in the market that are from the vault. So If vault assets are 80% of the total assets supplied `_vaultShare` will be 80 * 1e16